GoMoney/Common/InputView/CategoryPickerInputView.swift

```diff
@@ -0,0 +1,129 @@
+        let itemWidth = CGFloat((UIScreen.main.bounds.width - totalSpace) / CGFloat(numberOfItemsInSection))
+        let itemHeight = itemWidth
+
+        layout.sectionInset = UIEdgeInsets(top: spacing, left: spacing, bottom: spacing, right: spacing)
+        layout.minimumLineSpacing = spacing
+        layout.minimumInteritemSpacing = spacing
+        layout.itemSize = CGSize(width: itemWidth, height: itemHeight)
+
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.register(
+            CategoryPickerCell.self,
+            forCellWithReuseIdentifier: CategoryPickerCell.identifier
+        )
+        return collectionView
+    }()
+
+    var type: ExpenseType = .expense
+    var selectedTag: TransactionTag = expenseTags[0]
+
+    init(
+        type: ExpenseType = .expense,
+        didSelect: ((TransactionTag) -> Void)? = nil
+    ) {
+        super.init(frame: CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height: 200))
+        self.didSelect = didSelect
+        self.type = type
+        self.selectedTag = type == .expense ? expenseTags[0] : incomeTags[0]
+
+        setupView()
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupView() {
+        translatesAutoresizingMaskIntoConstraints = false
+        addSubview(collectionView)
+        collectionView.anchor(
+            top: topAnchor,
+            left: leftAnchor,
+            bottom: bottomAnchor,
+            right: rightAnchor
+        )
+    }
+
+    @objc func didChangeTheDate() {
+        didSelect?(selectedTag)
+    }
+}
+
+extension CategoryPickerInputView: UICollectionViewDelegate, UICollectionViewDataSource {
+    public func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return type == .expense ? expenseTags.count : incomeTags.count
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        if let cell = collectionView.dequeueReusableCell(withReuseIdentifier: CategoryPickerCell.identifier, for: indexPath) as? CategoryPickerCell {
+            let tag = type == .expense ? expenseTags[indexPath.row] : incomeTags[indexPath.row]
+            cell.transactionTag = tag
+            return cell
+        }
+        return UICollectionViewCell()
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let tag = type == .expense ? expenseTags[indexPath.row] : incomeTags[indexPath.row]
+        didSelect?(tag)
+    }
+}
+
+// MARK: Cell for collectionView
+
+class CategoryPickerCell: UICollectionViewCell {
```

Comment:
if just use in this file, should set `fileprivate`
else if use in another place => create another the file